### ISSUE TITLE
code has some quesetions

### ISSUE THREAD

**Issue by wangyazhao001** [YitongCU/Duel-staged-Attention-for-NYC-Weather-prediction#7]
def en_RNN(self,input_x):
        
        with tf.variable_scope('fw_lstm') as scope:
            cell = self.encode_cell  ## this step don't create variable 
            out, states = tf.nn.dynamic_rnn(cell, input_x, self.seq_len, dtype = tf.float32) ## this step does create variables
            
        tmp = tf.tile(states[1],[1,self.time_step])
        tmp = tf.reshape(tmp,[self.batch_size, self.time_step, self.n_hidden])
        
        concat = tf.concat([out,tmp],axis = 2)
        return concat ## shape shoule be (b,T,2*n_hidden)
#############################################################
In this code,you don't get the cell state of everystep!!!

**Comment by wangyazhao001** [comment]
You just got the hidden state because the states[1] is the  hidden state of last step.Instead,the states[0] is the cell state of the last step!just the last step!
